packages/react-dom/src/client/ReactDOMLegacy.js

```diff
@@ -265,78 +266,92 @@ export function hydrate(
   container: Container,
   callback: ?Function,
 ): React$Component<any, any> | PublicInstance | null {
-  if (__DEV__) {
-    console.error(
-      'ReactDOM.hydrate is no longer supported in React 18. Use hydrateRoot ' +
-        'instead. Until you switch to the new API, your app will behave as ' +
-        "if it's running React 17. Learn " +
-        'more: https://reactjs.org/link/switch-to-createroot',
+  if (disableLegacyReactDOMRenderAPIs) {
```

Comment:
I think we should move these errors up to ReactDOM.js, then the flag should be in the entry points. If the flag is on, they include these functions from ReactDOM. If it's off, then they import from ReactDOMLegacy. Would that work?